src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LoadedProject.cs

```diff
@@ -59,6 +59,15 @@ public LoadedProject(ProjectSystemProject projectSystemProject, SolutionServices
 
     private void FileChangedContext_FileChanged(object? sender, string filePath)
     {
+        var mostRecentIntermediateOutputPath = _mostRecentFileInfo?.IntermediateOutputPath;
+        if (mostRecentIntermediateOutputPath is not null
+            && PathUtilities.IsChildPath(mostRecentIntermediateOutputPath, filePath))
```

Comment:
I talked to Lifeng about this offline - the more correct solution here that devkit uses is to call the `project.GetAllGlobs` and related APIs on the MSBuild `Project` instance.  A `CompositeGlob` can be created from that which has a `IsMatch` API to determine if the file could impact the project.  e.g.
```
var d = new CompositeGlob(_loadedProject.GetAllGlobs().Select(r => r.MsBuildGlob).ToArray()); 
d.IsMatch("foo.cs");
```

This is problematic for us as the MSBuild APIs are only accessible from the build host process (by design).  The build host process has some limitations here -
1.  The build host process is currently short lived and only alive when we're loading the project.  By the time the file changes come in the build host process (and therefore the loaded MSBuild project instance) is gone.  
2.  As best as I can tell, there isn't a good way to for example serialize the underlying regex back to the host process so we can match against it there (we'd also have to re-implement the same input normalization that MSBuild does).

The most reliable solution to me seems to be to hold onto the buildhost process (and therefore the msbuild project instance) while the project is loaded.  But that seems to be a relatively significant rewrite of how the build host works - I'd like to wait until Jason is back to provide input before taking that approach.

Wanted to also tag @rainersigwald in case he had any brilliant ideas on ways to access the globs from a separate process after the MSBuild Project instance no longer is available.

Reply:
We could possibly add the globs to the project info sent back by the BuildHost. Using [Microsoft.Extensions.FileSystemGlobbing](https://learn.microsoft.com/en-us/dotnet/core/extensions/file-globbing) to avoid bringing MSBuild into the LSP Host project system (although at that point I am not sure why avoid the MSBuild dep).